src/source/Ice/SocketConnection.c

```diff
@@ -251,9 +252,10 @@ STATUS socketConnectionReadData(PSocketConnection pSocketConnection, PBYTE pBuf,
         // Unlikely that we will get sslReadRet == 0 here because socketConnectionReadData is only called when
         // socket recevies data. If ssl handshake is not done then -1 is returned.
         if (sslReadRet <= 0) {
-            sslErrorRet = SSL_get_error(pSocketConnection->pSsl, sslReadRet);
-            if (sslErrorRet != SSL_ERROR_WANT_WRITE && sslErrorRet != SSL_ERROR_WANT_READ) {
-                DLOGV("SSL_read failed with %s. Length of data already written %u", ERR_error_string(sslErrorRet, NULL), writtenBytes);
+            while ((sslErrorRet = ERR_get_error()) != 0) {
```

Comment:
Perhaps deserves a comment, not sure what this returns

Reply:
This is just the way openssl error is printed.

Reply:
This really does look fishy. I don't see anything in the description requiring to iterate: https://www.openssl.org/docs/manmaster/man3/ERR_get_error.html

Reply:
the way it mentioned in the doc doesnt give the proper error.